src/olympia/blocklist/cron.py

```diff
@@ -68,72 +69,78 @@ def _upload_mlbf_to_remote_settings(*, force_base=False):
     # This timestamp represents the last time the MLBF was generated and uploaded.
     # It could have been a base filter or a stash.
     last_generation_time = get_last_generation_time()
-    # This timestamp represents the point in time when
-    # the base filter was generated and uploaded.
-    base_generation_time = get_base_generation_time()
-
     mlbf = MLBF.generate_from_db(generation_time)
-
-    base_filter = (
-        MLBF.load_from_storage(base_generation_time)
-        if base_generation_time is not None
-        else None
-    )
-
     previous_filter = (
-        # Only load previoous filter if there is a timestamp to use
-        # and that timestamp is not the same as the base_filter
         MLBF.load_from_storage(last_generation_time)
         if last_generation_time is not None
-        and (base_filter is None or base_filter.created_at != last_generation_time)
-        else base_filter
+        else None
     )
 
-    changes_count = mlbf.blocks_changed_since_previous(
-        BlockType.BLOCKED, previous_filter
-    )
-    statsd.incr(
-        'blocklist.cron.upload_mlbf_to_remote_settings.blocked_changed', changes_count
+    base_filters_to_update: List[BlockType] = []
+    update_stash = False
+
+    # Determine which base filters need to be re uploaded
+    # and whether the stash needs to be updated
+    for block_type in BlockType:
+        base_generation_time = get_base_generation_time(block_type)
+        base_filter = (
+            MLBF.load_from_storage(base_generation_time)
+            if base_generation_time is not None
+            else None
+        )
+
+        update_filter = (
+            force_base
+            or base_filter is None
+            or mlbf.blocks_changed_since_previous(block_type, base_filter)
+            > BASE_REPLACE_THRESHOLD
+        )
+
+        if update_filter:
+            base_filters_to_update.append(block_type)
+        elif (
+            mlbf.blocks_changed_since_previous(
+                block_type, previous_filter or base_filter
+            )
+            > 0
+        ):
+            update_stash = True
+
+    previous_filter_is_stale = (
+        previous_filter is not None
+        and previous_filter.created_at < get_blocklist_last_modified_time()
     )
+
     need_update = (
-        force_base
```

Comment:
So we cannot use `force_base` anymore?

Reply:
Still [there](https://github.com/mozilla/addons-server/pull/22828/files#diff-2c1350128209aae0e622bd63c101e306f735752352fd0d5d8b8447785bd3a56fR93), but somewhere else in the code. There are tests verifying force_base recreates filters even if any filter should not have otherwise been recreated.